### PR TITLE
Add FP repro for #7058

### PR DESCRIPTION
**Pull request by antonioaversa**
Reproduces #7058

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs [hunk 1]
```diff
@@ -214,3 +214,16 @@ namespace ANamespace
...
+// https://github.com/SonarSource/sonar-dotnet/issues/7058
+class Repro_7058
+{
+    void Test(int[] a)
+    {
+        var local = new object();
+        Action action = () =>
+        {
+            lock (local) { } // Noncompliant, FP: local is in the captured scope
+        };
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs:227]: Add a counter example where there is an action, local is captured, but the lock is outside the delegate:
```cs
        var local = new object();
        Action action = () =>
        {
            local.ToString();
        };
        lock (local) { } // Noncompliant
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs [hunk 2]
```diff
@@ -214,3 +214,21 @@ namespace ANamespace
 {
     class AClass { }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7058
+class Repro_7058
+{
+    void Test(int[] a)
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs:221]:
```suggestion
    void LockInsideCapture()
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs [hunk 3]
```diff
@@ -214,3 +214,21 @@ namespace ANamespace
...
+        Action action1 = () =>
+        {
+            lock (local) { } // Noncompliant, FP: local is in the captured scope
+        };
+        Action action2 = () =>
+        {
+            var localCaptured = local;
+        };
+        lock (local) { }     // Noncompliant, locking captured variable in its declaration scope
+    }
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs:233]:
```suggestion
    }

    void LockOutsideCapture()
    {
        var local = new object();
        Action action = () =>
        {
            var localCaptured = local;
        };
        lock (local) { }     // Noncompliant, locking captured variable in its declaration scope
    }
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs (hunk 3 above)
Comment (antonioaversa) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs]: Changed.
Splitting into two methods and having specific names is, however, in this scenario and in my opinion, of very little help.
The non-compliance comment, which is right next to the primary position by importance, explains well the nature of the non-compliance. Repeating it 3-4 lines away from that location, in the name of the method, is not useful.
Moreover, it should be considered nitpicking and not blocking for the PR.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**